### PR TITLE
Add API for getting View from SeqNo

### PR DESCRIPTION
**Pull request by eddyashton**
Rather than accessing `consensus` directly, we want to provide a stable API for apps to access this common functionality. This PR adds `get_view_for_seqno_v1`, and an endpoint in `nobuiltins` which uses it.

### CODE REVIEW

File: samples/apps/nobuiltins/nobuiltins.cpp [hunk 1]
```diff
@@ -131,6 +133,67 @@ namespace nobuiltins
...
+            }
+
+            kv::Consensus::View view;
+            const auto result = get_view_for_seqno_v1(seqno, view);
+            if (result == ccf::ApiResult::OK)
+            {
+              ccf::TxID tx_id;
+              tx_id.view = view;
+              tx_id.seqno = seqno;
+              nlohmann::json out = nlohmann::json::object();
```
Comment (jumaffre) [inline samples/apps/nobuiltins/nobuiltins.cpp:167]: Why not using an out schema for this endpoint too?

**Review by jumaffre — APPROVED** [review]
Could you add a quick end-to-test for this new endpoint?

File: samples/apps/nobuiltins/nobuiltins.cpp (hunk 1 above)
Comment (eddyashton) [reply samples/apps/nobuiltins/nobuiltins.cpp]: No real reason, I've added this now.

**Comment by eddyashton** [comment]
> Could you add a quick end-to-test for this new endpoint?

We actually didn't have an e2e test for this nobuiltins app at all, but I've added it now.

File: tests/nobuiltins.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    with infra.network.network(
+        args.nodes, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
+    ) as network:
+        network.start_and_join(args)
+
+        test_nobuiltins_endpoints(network, args)
+
+
+if __name__ == "__main__":
+    LOG.warning("Hello")
```
Comment (achamayou) [inline tests/nobuiltins.py:48]: This line and line 51 are presumably temporary?

**Review by achamayou — APPROVED** [review]

File: tests/nobuiltins.py (hunk 2 above)
Comment (eddyashton) [reply tests/nobuiltins.py]: Haha, yes! Removed.

**Pull request merged**